spyder_unittest/unittestplugin.py

```diff
@@ -24,16 +26,67 @@
 _ = get_translation("unittest", dirname="spyder_unittest")
 
 
-class UnitTestPlugin(UnitTestWidget, SpyderPluginMixin):
-    """Unit testing."""
+class UnitTestWidgetInSpyder(UnitTestWidget):
+    """
+    Unit test widget for use inside Spyder.
+
+    This class overrides some functions in `UnitTestWidget` to provide better
+    integration with Spyder.
+    """
+
+    def __init__(self, main):
+        """
+        Constructor.
+
+        Parameters
+        ----------
+        main : MainWindow
+            Spyder main window
+        """
+        UnitTestWidget.__init__(self, parent=main)
```

Comment:
I thought a bit more about this.

The main problem is pythonpath. This plugin, and also the profiler and line_profiler plugin, need to have the pythonpath set correctly.

I can see these possibilities:

1. The widget has a reference to the main window. This is against the current idea of the architecture.
2. The widget has a reference to the plugin, which has a reference to the main window. I suppose this is also not allowed.
3. The Spyder application emits a signal whenever the pythonpath is changed with the new value, and the plugin connects the widget to the signal. This is similar to how changes in the configuration are currently handled (I believe). This is quite a clean design, but it requires cooperation from the Spyder application, and there may be performance issues if the Spyder application has to emit signals whenever something happens that some plugin may be interested in (I don't think this is a real issue at the moment).
4. The widget has to do without the value of pythonpath. When the user clicks "Start unit tests" from the menu, this is handled by the plugin so the correct value of pythonpath can be injected. When the user clicks the "Start" button in the widget, it is the responsibility of the user to set the pythonpath correctly (so a suitable window should pop up). I would find it annoying that the "Start" button in the menu and the "Start" button in the widget behave differently.

@ccordoba12 @goanpeca What do you think? Any other suggestions?